File: math-functions/src/main/java/io/github/svbgabriel/FunctionResource.java
package io.github.svbgabriel;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import org.jboss.resteasy.reactive.RestQuery;

@Path("/api/functions")
public class FunctionResource {

    @Inject
    FunctionService functionService;

    @GET
    @Path("factorial")
    @Produces(MediaType.APPLICATION_JSON)
    public ResultFactorialBody factorial(@RestQuery("input") final Long value) {
        return functionService.factorial(value);
    }

    @GET
    @Path("fibonacci")
    @Produces(MediaType.APPLICATION_JSON)
    public ResultFibonacciBody fibonacci(@RestQuery("input") final Long value) {
        return functionService.fibonacci(value);
    }
}

File: math-functions/src/main/kotlin/io/github/svbgabriel/FunctionResource.kt
package io.github.svbgabriel

import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType
import org.jboss.resteasy.reactive.RestQuery

@Path("/api/functions")
class FunctionResource(private val functionService: FunctionService) {
    @GET
    @Path("factorial")
    @Produces(MediaType.APPLICATION_JSON)
    fun factorial(@RestQuery("input") value: Long): ResultFactorialBody {
        return functionService.factorial(value)
    }

    @GET
    @Path("fibonacci")
    @Produces(MediaType.APPLICATION_JSON)
    fun fibonacci(@RestQuery("input") value: Long): ResultFibonacciBody {
        return functionService.fibonacci(value)
    }
}
